Tasks/DownloadBuildArtifacts/task.json

```diff
@@ -30,30 +30,55 @@
       "type": "pickList",
       "label": "Build definition",
       "defaultValue": "",
-      "required": true,
+      "required": false,
       "properties": {
         "EditableOptions": "True"
       },
       "helpMarkDown": "Select the build definition name"
     },
     {
-      "name": "version",
+      "name": "buildId",
       "type": "pickList",
       "label": "Build",
-      "defaultValue": "",
+      "defaultValue": "$(Build.BuildId)",
       "required": true,
       "properties": {
         "EditableOptions": "True"
       },
       "helpMarkDown": "Select the build number"
     },
     {
+      "name": "downloadType",
+      "type": "radio",
+      "label": "Download Type",
+      "defaultValue": "single",
+      "required": true,
+      "helpMarkDown": "Download specific or multiple artifacts matching pattern.",
+      "options": {
+          "single": "Single Artifact",
+          "multiple": "All Artifacts"
+      }
+    },
+    {
+      "name": "artifactName",
+      "type": "pickList",
+      "label": "Artifact name",
+      "defaultValue": "drop",
+      "required": true,
+      "properties": {
+        "EditableOptions": "True"
+      },
+      "visibleRule": "downloadType == single",
+      "helpMarkDown": "The name of the artifact to download"
+    },
+    {
       "name": "itemPattern",
       "type": "string",
-      "label": "Item Pattern",
+      "label": "Matching Pattern",
       "defaultValue": "**",
       "required": false,
-      "helpMarkDown": "Minmatch pattern to filter files to be downloaded."
+      "visibleRule": "downloadType == multiple",
+      "helpMarkDown": "Minimatch pattern to filter files to be downloaded."
```

Comment:
Help text needs to be include an example. Should the name of the artifact be included in the pattern? For example drop/foo/*?

Reply:
modified the help text.